Client/mods/deathmatch/logic/CClient2DFXManager.cpp

```diff
@@ -0,0 +1,1007 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        Client/mods/deathmatch/logic/CClient2DFXManager.cpp
+ *  PURPOSE:     Handling 2DFX effects manager
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+#include "CClient2DFXManager.h"
+
+#define mask(n) ((1 << (n)) - 1)
+static constexpr std::size_t roadsignTextSize = 64;
+
+CClient2DFXManager::CClient2DFXManager(CClientManager* mainManager) : m_mainManager(mainManager)
+{
+    // Init
+    m_canRemoveFromList = true;
+}
+
+CClient2DFXManager::~CClient2DFXManager()
+{
+    // Delete all 2DFX's
+    RemoveAll();
+}
+
+void CClient2DFXManager::RemoveAll()
+{
+    // Make sure they don't remove themselves from our list
+    m_canRemoveFromList = false;
+
+    // Run through our list deleting the 2DFX's
+    for (auto* effect : m_effectsList)
+        delete effect;
```

Comment:
dont you also need to remove the elements from the list?

Reply:
This function is only called in the destructor, so it is not necessary. Other managers don't have this either